pyscript.core/src/stdlib/pyscript/event_handling.py

```diff
@@ -41,8 +41,16 @@ def decorator(func):
             # Function doesn't receive events
             if not sig.parameters:
 
-                def wrapper(*args, **kwargs):
-                    func()
+                # Function is async: must be awaited
+                if inspect.iscoroutinefunction(func):
```

Comment:
Interestingly, this bit of code only ever runs in Pyodide (see the ugly hack comment elsewhere in this function).

Reply:
just in case it was literally in this MR description:

> In Pyodide (only) use case when a function has no args but it's async, the wrapper is not asynchronous and it doesn't await the passed callback. This results into hard to understand errors otherwise not present in MicroPython.

😄